README.md

```diff
@@ -1,2 +1,429 @@
+        <p><strong>Customer:</strong> {{ Model.CustomerName }}</p>
+        <p><strong>Date:</strong> {{ Model.Date | date.to_string '%d/%m/%Y' }}</p>
+    </div>
+    
+    <table>
+        <thead>
+            <tr>
+                <th>Item</th>
+                <th>Quantity</th>
+                <th>Price</th>
+                <th>Total</th>
+            </tr>
+        </thead>
+        <tbody>
+            {{- for item in Model.Items }}
+            <tr>
+                <td>{{ item.Name }}</td>
+                <td>{{ item.Quantity }}</td>
+                <td>{{ item.Price | object.format "C" }}</td>
+                <td>{{ item.Quantity * item.Price | object.format "C" }}</td>
+            </tr>
+            {{- end }}
+        </tbody>
+        <tfoot>
+            <tr>
+                <td colspan="3" class="total">Total:</td>
+                <td class="total">{{ Model.Total | object.format "C" }}</td>
+            </tr>
+        </tfoot>
+    </table>
+</body>
+</html>
+""";
+
+var request = new PdfGenerationRequest(htmlTemplate, order, templateEngine: "scriban");
+```
+
+## ⚠️ Error Handling
+
+The API returns appropriate HTTP status codes and error details:
+
+### Common Status Codes
+
+- **200 OK**: PDF generated successfully
+- **400 Bad Request**: Invalid request data or template errors
+- **401 Unauthorized**: Invalid or missing API key
+- **408 Request Timeout**: Generation took longer than 30 seconds
+- **429 Too Many Requests**: Rate limit exceeded
+- **500 Internal Server Error**: Unexpected server error
+
+### Error Response Format
+
+```json
+{
+  "type": "https://httpstatuses.com/400",
+  "title": "Bad Request",
+  "status": 400,
+  "detail": "Template engine 'invalid' has not been registered",
+  "instance": "/api/pdf"
+}
+```
+
+## 🛡️ Rate Limiting
+
+Rate limiting is enforced per subscription:
+
+- **Window-based limiting**: Requests are counted within specified time windows
+- **Per-user limits**: Each API key has its own rate limit configuration
+- **429 Response**: When limits are exceeded, includes `Retry-After` header
+- **Configurable**: Administrators can set custom limits per subscription
+
+Example rate limit headers:
+```
+Retry-After: 60
+```
+
+## ⚙️ Configuration
+
+### Application Settings
+
+Key configuration options in `appsettings.json`:
+
+```json
+{
+  "ConnectionStrings": {
+    "SqlConnection": "Server=.;Database=PdfSmith;Trusted_Connection=true;"
+  },
+  "AppSettings": {
+    "AdministratorUserName": "admin",
+    "AdministratorApiKey": "your-admin-api-key"
+  },
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
+      "Microsoft.AspNetCore": "Warning"
+    }
+  }
+}
+```
+
+### Environment Variables
+
+- `PLAYWRIGHT_BROWSERS_PATH`: Custom path for Playwright browsers installation
+
+### Playwright Configuration
+
+Chromium is automatically installed via Playwright. On Windows, browsers are installed to:
+```
+%USERPROFILE%\AppData\Local\ms-playwright
```

Comment:
Add default location for Linux and Mac and specify that this locations are used if PLAYWRIGHT_BROWSERS_PATH isn't specified